### PR TITLE
closes #14 adtte vignette template

### LINKED ISSUE #14: ADTTE vignette & template

### PR DESCRIPTION
**Pull request by rossfarrugia**
Closes #14 .

Thank you for your Pull Request! We have developed this task checklist from the [Development Process Guide](https://pharmaverse.github.io/admiral/articles/development_process.html) to help with the final steps of the process. Completing the below tasks helps to ensure our reviewers can maximize their time on your code as well as making sure the admiral codebase remains robust and consistent.   

Please check off each taskbox as an acknowledgment that you completed the task or check off that it is not relevant to your Pull Request. This checklist is part of the Github Action workflows and the Pull Request will not be merged into the `devel` branch until you have checked off each task.

- [x] Code is formatted according to the [tidyverse style guide](https://style.tidyverse.org/) 
- [x] Updated relevant unit tests or have written new unit tests - See [Unit Test Guide](https://pharmaverse.github.io/admiral/articles/unit_test_guidance.html#writing-unit-tests-in-admiral-)
- [x] If you removed/replaced any function and/or function parameters, did you fully follow the [deprecation guidance](https://pharmaverse.github.io/admiral/articles/programming_strategy.html#deprecation-1)?
- [x] Update to all relevant roxygen headers and examples 
- [x] Run `devtools::document()` so all `.Rd` files in the `man` folder and the `NAMESPACE` file in the project root are updated appropriately
- [x] Address any updates needed for vignettes and/or templates
- [x] Update `NEWS.md`
- [x] Build admiral site `pkgdown::build_site()` and check that all affected examples are displayed correctly and that all new functions occur on the "[Reference](https://pharmaverse.github.io/admiralonco/reference/index.html)" page. 
- [x] Address or fix all lintr warnings and errors - `lintr::lint_package()`
- [x] Run `R CMD check` locally and address all errors and warnings - `devtools::check()`
- [x] Link the issue so that it closes after successful merging. 
- [x] Address all merge conflicts and resolve appropriately 
- [x] Pat yourself on the back for a job well done!  Much love to your accomplishment!

### CODE REVIEW

**Comment by rossfarrugia** [comment]
First versions of the ADTTE vignette and template available in case anyone wants to do some early review. As I can't yet create an ADRS (waiting on last few functions) I tested the code using a simple test ADRS generted as follows:

```
adrs <- tibble::tribble(
  ~STUDYID,       ~USUBJID,           ~ADTC,        ~PARAMCD, ~AVALC, ~ANL01FL, ~ASEQ,
  "CDISCPILOT01", "01-701-1015",      "2014-07-02", "DEATH",  "Y",    "Y",      "1",
  "CDISCPILOT01", "01-701-1015",      "2014-06-01", "PD",     "Y",    "Y",      "2",
  "CDISCPILOT01", "01-701-1015",      "2014-06-01", "LSTA",   "Y",    "Y",      "3",
  "CDISCPILOT01", "01-701-1015",      "2014-05-01", "RSP",    "Y",    "Y",      "4",
  "CDISCPILOT01", "01-701-1023",      "",           "DEATH",  "N",    "Y",      "1",
  "CDISCPILOT01", "01-701-1023",      "",           "PD",     "N",    "Y",      "2",
  "CDISCPILOT01", "01-701-1023",      "2012-09-01", "LSTA",   "Y",    "Y",      "3",
  "CDISCPILOT01", "01-701-1023",      "2012-08-13", "RSP",    "Y",    "Y",      "4"
) %>%
    mutate(
      ADT = lubridate::ymd(ADTC)
    ) %>%
    derive_vars_merged(
      dataset_add = adsl,
      by_vars = vars(STUDYID, USUBJID),
      new_vars = vars(TRTSDT, LSTALVDT)
    ) %>%
  mutate(
    RANDDT = TRTSDT, RSDTC=ADTC
  )
```

**Still need to add `data/admiral_adtte.rda` created using the template once we have ADRS as input.**

File: vignettes/adtte.Rmd [hunk 1]
```diff
@@ -0,0 +1,265 @@
...
+The events and censorings are defined by the `admiral::event_source()` and the
+`admiral::censor_source()` class respectively. It defines
+
+- which observations (`filter` parameter) of a source dataset (`dataset_name`
+parameter) are potential events or censorings,
+- the value of the `CNSR` variable (`censor` parameter), and
+- which variable provides the date (`date` parameter).
+
+The date can be provided as date (`--DT` variable), datetime (`--DTM` variable),
+or character ISO-8601 date (`--DTC` variable).
```
Comment (thomas-neitmann) [inline vignettes/adtte.Rmd:96]: We decided to not accept `--DTC` variabls going forward so this will have to be updated.

File: vignettes/adtte.Rmd [hunk 2]
```diff
@@ -0,0 +1,265 @@
...
+the `dataset_name` parameter. This allows to define events and censoring
+independent of the data.
+
+### Pre-Defined Time-to-Event Source Objects
+
+The table below shows all pre-defined `tte_source` objects which should cover the most
+common oncology use cases.
+
+```{r echo=FALSE}
+knitr::kable(admiral:::list_tte_source_objects(package = "admiralonco"))
```
Comment (thomas-neitmann) [inline vignettes/adtte.Rmd:115]: No need for `:::` as `list_tte_sources()` is now an exported function of {admiral}.

File: vignettes/adtte.Rmd (hunk 1 above)
Comment (rossfarrugia) [reply vignettes/adtte.Rmd:96]: are you saying this is something to update now before the testing phase starts next month? or you're giving a heads up later in line with once admiral core makes this update? (as i copied this bit from admiral core vignette - so probably makes sense one person updates this across the packages)

p.s. i think the admiral core partial date imputation refactor i proposed on our Box will cause such impact in many places, so this will be one big update.

Comment (thomas-neitmann) [reply vignettes/adtte.Rmd:96]: I think we can keep it as it for now but will have to revisit once the changes to ate imputation have been made.

Comment (rossfarrugia) [reply vignettes/adtte.Rmd:96]: agree! will resolve for now and we'll update this across all the admiral family of packages later

**Comment by rossfarrugia** [comment]
NOTE: The reason this PR doesn't also store an `admiral_adtte.rda` at `data/` is the admiral core team leads recently agreed a strategy change to only store any ADaMs where vignettes etc depend on these (like ADRS) as otherwise the packages get bogged down with large datasets.

Here's a to-do list of remaining actions for this PR:

[ ] ADTTE template will need to be re-ran and sanity checked with the latest `admiral_adrs` input once PR #54 completed, as currently ADTTE looks a bit bare due to the missing RANDDT issue.

[ ]  Build site to check the vignette renders - I'm having no luck currently with getting this working

[ ]  Fix the remaining R-CMD check and Docs auto-checks

**Review by bundfussr — CHANGES_REQUESTED** [review]

**Comment by bundfussr** [comment]
Checks are fixed in https://github.com/pharmaverse/admiralonco/pull/54.

File: vignettes/adtte.Rmd [hunk 3]
```diff
@@ -0,0 +1,265 @@
...
+    censor_conditions = list(lasta_censor, rand_censor),
+    source_datasets = list(adsl = adsl, adrs = adrs),
+    set_values_to = vars(PARAMCD = "PFS", PARAM = "Progression Free Survival")
+  ) %>%
+  derive_param_tte(
+    dataset_adsl = adsl,
+    start_date = TEMP_RESPDT,
+    event_conditions = list(pd_event, death_event),
+    censor_conditions = list(lasta_censor),
+    source_datasets = list(adsl = filter(adsl, !is.na(TEMP_RESPDT)), adrs = adrs),
```
[the remark this answers is not in the document]
Comment (rossfarrugia) [reply vignettes/adtte.Rmd:184]: are you sure on this one @bundfussr? when i ran with some test data i thought it had worked as the patient needs to have a response in ADSL and then this gets merged with the ADRS and so those patients with no response are removed. can you double check with some test data and let me know if i need to update, maybe you'll spot something i missed.

[the remark this answers is not in the document]
Comment (rossfarrugia) [reply vignettes/adtte.Rmd:183]: @bundfussr ok i will update so this is more robust, but can you see any reason why the same `derive_param_tte` call works for me in my ADTTE template with our test data?? e.g. see the following where patients 1023 and 1033 are excluded as they had no TEMP_RESPDT:

image

seems odd!

[the remark this answers is not in the document]
Comment (rossfarrugia) [reply vignettes/adtte.Rmd]: i think it was that those patients had no original ADRS records so they were somehow getting stripped out anyway by coincidence. i've updated the code now and works. just need to run through the auto checks and think wordlist will have a merge conflict i need to resolve. then will request a re-review

**Comment by rossfarrugia** [comment]
@bundfussr i'm doing some last checks on my side but think this is ready for a re-review now

File: vignettes/adtte.Rmd [hunk 4]
```diff
@@ -0,0 +1,272 @@
...
+dataset_vignette(
+  adtte,
+  display_vars = vars(USUBJID, PARAMCD, PARAM, STARTDT, ADT, CNSR)
+)
+```
+
+### Creating Your Own Time-to-Event Source Objects
+
+We advise you consult the `{admiral}` ["Creating a BDS Time-to-Event ADaM" vignette]
+(https://pharmaverse.github.io/admiral/articles/bds_tte.html) for further guidance on the
```
Comment (bundfussr) [inline vignettes/adtte.Rmd:205]: The link to the vignette does not render correctly. Maybe there must be no line break between link name and URL.

File: vignettes/adtte.Rmd [hunk 5]
```diff
@@ -0,0 +1,272 @@
...
+
+### Pre-Defined Time-to-Event Source Objects
+
+The table below shows all pre-defined `tte_source` objects which should cover the most
+common oncology use cases.
+
+```{r echo=FALSE}
+knitr::kable(list_tte_source_objects(package = "admiralonco"))
+```
+
```
Comment (bundfussr) [inline vignettes/adtte.Rmd:117]: I wonder if we should remove `ANL01FL == "Y"` from the conditions of the `tte_source` objects. For parameters like PD, DEATH and LASTA it does not make much sense as there is just one observation per subject, i.e., it is fulfilled anyway.

File: vignettes/adtte.Rmd [hunk 6]
```diff
@@ -0,0 +1,272 @@
...
+to filter the input datasets so as to only derive the records for responders.
+This could also be repeated as needed for IRF and confirmed responses.
+
+Here is an example of the code needed.
+
+```{r}
+adsl <- adsl %>%
+  derive_vars_merged(
+    dataset_add = adrs,
+    filter_add = PARAMCD == "RSP" & AVALC == "Y" & ANL01FL == "Y",
```
Comment (bundfussr) [inline vignettes/adtte.Rmd:150]: I would remove `ANL01FL == "Y"` here as well.

File: inst/templates/ad_adtte.R [hunk 7]
```diff
@@ -0,0 +1,96 @@
...
+adsl <- admiral_adsl
+adrs <- admiral_adrs
+
+# ---- Derivations ----
+
+# Add response date to ADSL / ADRS for duration of response calculation
+adsl <- adsl %>%
+  derive_vars_merged(
+    dataset_add = adrs,
+    filter_add = PARAMCD == "RSP" & AVALC == "Y" & ANL01FL == "Y",
```
Comment (bundfussr) [inline inst/templates/ad_adtte.R:30]: I would remove `ANL01FL == "Y"`.

**Review by bundfussr — CHANGES_REQUESTED** [review]

File: vignettes/adtte.Rmd (hunk 5 above)
Comment (rossfarrugia) [reply vignettes/adtte.Rmd:117]: technically you're right but i prefer to keep them in for consistency and reminding of this important consideration of the ADRS workflow, plus user could add ANL02FL / ANLxxFL etc with different scenarios or for periods so it would remind them to consider this

File: vignettes/adtte.Rmd (hunk 6 above)
Comment (rossfarrugia) [reply vignettes/adtte.Rmd:150]: see my explanation above

File: inst/templates/ad_adtte.R (hunk 7 above)
Comment (rossfarrugia) [reply inst/templates/ad_adtte.R:30]: see my explanation above

File: vignettes/adtte.Rmd (hunk 4 above)
Comment (rossfarrugia) [reply vignettes/adtte.Rmd]: @bundfussr i tried the fix you suggested - can you let me know if works? i'm having trouble with my R session so not able to build the site from this branch to check

Comment (bundfussr) [reply vignettes/adtte.Rmd]: It is working now.

File: vignettes/adtte.Rmd (hunk 5 above)
Comment (bundfussr) [reply vignettes/adtte.Rmd:117]: If you have `ANL01FL` for selecting the worst assessment per date and `ANL02FL` for selecting assessments prior anti cancer therapy for the OVR parameter, would you set both variables for other parameters like PD, RSP, or DEATH? And would you always use both variables when selecting observations for PD, RSP, or DEATH?

Comment (rossfarrugia) [reply vignettes/adtte.Rmd:117]: i think in that scenario users would probably create new parameters - e.g. you'd have a PARAMCD=PD with ANL01FL=Y record and also say a PARAMCD=PDNACT with ANL02FL=Y. as you could have a patient who is a PD in the first condition but not in the 2nd, as the PD occurred after NACT (next anti cancer therapy). again your point would still stand as the ANLxxFLs wouldn't add anything to these records, except it just gives a more visual tie to the source records that contributed to each parameter, so would be appreciated i'm sure by an FDA reviewer.

the other more complex scenario (but likely rare) would be efficacy by period/phase - where you might have one PD record but ANLxxFL would get set dependent on which period/phase the event occurred at. in this case if you then needed to downstream do say a "time to event during neo-adjuvant phase" output then filtering with the right ANLxxFL would be critical.

in a nutshell, i agree with your comment but i rather leave it in as it does little harm and it opens up to users to consider possibly more complex protocol-specific needs.

**Review by bundfussr — APPROVED** [review]
